### PR TITLE
Apply BEM refactor for drawer

### PR DESCRIPTION
**Pull request by AlexanderSchmidtCE**
Ticket: #7036 

The A11y and PO reviews will only take place after all other DoD steps have been completed by the Developer:
- [x] Meaningful pull request title for the release notes
- [x] Pull request is linked to an issue and all changes relate to the issue
- [x] Tests to protect this code implemented (if applicable)
- [x] Manual test performed successfully (if applicable)
- [x] Documentation or migration has been updated (if applicable)

### CODE REVIEW

File: packages/components/src/components/drawer/shadow.tsx [hunk 1]
```diff
@@ -55,10 +55,10 @@ export class KolDrawer implements DrawerAPI {
 		return (
 			<div
 				ref={this.getWrapperRef}
-				class={`drawer__wrapper drawer__wrapper--${align} ${this.state._open ? 'drawer__wrapper--open' : 'is-closing'}`}
+				class={`kol-drawer__wrapper kol-drawer__wrapper--${align} ${this.state._open ? 'kol-drawer__wrapper--open' : 'kol-drawer__wrapper--is-closing'}`}
```
Comment (sdvg) [inline packages/components/src/components/drawer/shadow.tsx]: Hier würde sich `clsx` anbieten

File: packages/components/src/components/drawer/shadow.tsx [hunk 2]
```diff
@@ -69,8 +69,8 @@ export class KolDrawer implements DrawerAPI {
 	public render(): JSX.Element {
 		const isModal = this.state._modal;
 		return (
-			<Host class={`kol-drawer drawer ${isModal ? 'drawer--modal' : ''}`} ref={(el) => (this.hostElement = el as HTMLElement)}>
-				<dialog class="drawer__dialog" ref={this.getRef}>
+			<Host class={`kol-drawer ${isModal ? 'kol-drawer--modal' : ''}`} ref={(el) => (this.hostElement = el as HTMLElement)}>
```
Comment (sdvg) [inline packages/components/src/components/drawer/shadow.tsx]: clsx

File: packages/components/src/components/drawer/style.scss [hunk 3]
```diff
@@ -6,7 +6,9 @@
 		font-size: rem(16);
 	}
```
Comment (sdvg) [inline packages/components/src/components/drawer/style.scss]: Standardfrage: Brauchen wir die font-size auf :host und root-Element?

Comment (AlexanderSchmidtCE) [reply packages/components/src/components/drawer/style.scss]: hab vergessen gehabt den root-teil zu entfernen

**Review by sdvg — DISMISSED** [review]

**Review by sdvg — APPROVED** [review]

**Pull request merged**